From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.15 october2005, patch 02-29

  Description: zfcp: correct this_id of zfcp hosts
  Symptom:     No scsi_target is created for the first remote port that
               is registered for the zfcp host when this_id equals 0.
  Problem:     this_id should not be set to 0 in the scsi_host_template of
               zfcp hosts.
  Solution:    Correctly initialize this_id with -1 in scsi_host_template.
  Problem-ID:  -
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/scsi/zfcp_scsi.c	8 Mar 2006 20:09:37 -0000	1.101.2.5
+++ linux-2.5/drivers/s390/scsi/zfcp_scsi.c	9 Mar 2006 11:29:17 -0000	1.101.2.6
@@ -70,7 +70,7 @@
 	      eh_host_reset_handler:   zfcp_scsi_eh_host_reset_handler,
 			               /* FIXME(openfcp): Tune */
 	      can_queue:               4096,
-	      this_id:	               0,
+	      this_id:	               -1,
 	      /*
 	       * FIXME:
 	       * one less? can zfcp_create_sbale cope with it?

